exporter/elasticsearchexporter/factory.go

```diff
@@ -116,10 +116,13 @@ func createTracesExporter(ctx context.Context,
 		exporterhelper.WithQueue(cf.QueueSettings))
 }
 
-// set default User-Agent header
-func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) {
-	if cf.Headers == nil {
-		cf.Headers = make(map[string]string)
+// set default User-Agent header if User-Agent is empty
+func setDefaultUserAgentHeader(cf *Config, info component.BuildInfo) *Config {
```

Comment:
I'm a bit confused by the combination of a method name `set...` while including a return here: I'd expect a method to either return an updated value, or set the value itself, not both. Is there something I'm missing or can we simplify the method here? 

In other words, the returned value is simply the passed in `cf` variable, so I don't see the point of returning anything.

Reply:
make sense, there are some confusing semantics